src/sgp/VSurface.cc

```diff
@@ -304,123 +313,38 @@ void ColorFillVideoSurfaceArea(SGPVSurface* const dst, INT32 iDestX1, INT32 iDes
 // Will drop down into user-defined blitter if 8->16 BPP blitting is being done
 void BltVideoSurface(SGPVSurface* const dst, SGPVSurface* const src, INT32 const iDestX, INT32 const iDestY, SGPBox const* const src_box)
 {
+	// maxrd2 - OK - used on intro screen, main menu, after load game screen
 	Assert(dst);
+	Assert(dst->BPP() == 32);
 	Assert(src);
+	Assert(src->BPP() == 32);
 
 	const UINT8 src_bpp = src->BPP();
 	const UINT8 dst_bpp = dst->BPP();
-	if (src_bpp == dst_bpp)
-	{
-		SDL_Rect* src_rect = 0;
-		SDL_Rect  r;
-		if (src_box)
-		{
-			r.x = src_box->x;
-			r.y = src_box->y;
-			r.w = src_box->w;
-			r.h = src_box->h;
-			src_rect = &r;
-		}
-
-		SDL_Rect dstrect;
-		dstrect.x = iDestX;
-		dstrect.y = iDestY;
-		SDL_BlitSurface(src->surface_, src_rect, dst->surface_, &dstrect);
-	}
-	else if (src_bpp < dst_bpp)
-	{
-		SGPBox const* src_rect = src_box;
-		SGPBox        r;
-		if (!src_rect)
-		{
-			// Check Sizes, SRC size MUST be <= DEST size
-			if (dst->Height() < src->Height())
-			{
-				SLOGD(DEBUG_TAG_VSURFACE, "Incompatible height size given in Video Surface blit");
-				return;
-			}
-			if (dst->Width() < src->Width())
-			{
-				SLOGD(DEBUG_TAG_VSURFACE, "Incompatible height size given in Video Surface blit");
-				return;
-			}
-
-			r.x = 0;
-			r.y = 0;
-			r.w = src->Width();
-			r.h = src->Height();
-			src_rect = &r;
+	if(src_bpp == dst_bpp) {
```

Comment:
This check is now redundant — either remove or move up, so it has a chance to log.